pkg/cluster/storage.go

```diff
@@ -10,23 +10,25 @@ import (
 	"strings"
 
 	"github.com/k3s-io/kine/pkg/client"
+	"github.com/k3s-io/kine/pkg/endpoint"
 	"github.com/rancher/k3s/pkg/bootstrap"
 	"github.com/rancher/k3s/pkg/clientaccess"
+	"github.com/rancher/k3s/pkg/daemons/config"
 	"github.com/sirupsen/logrus"
 )
 
-// save writes the current ControlRuntimeBootstrap data to the datastore. This contains a complete
+// Save writes the current ControlRuntimeBootstrap data to the datastore. This contains a complete
 // snapshot of the cluster's CA certs and keys, encryption passphrases, etc - encrypted with the join token.
 // This is used when bootstrapping a cluster from a managed database or external etcd cluster.
 // This is NOT used with embedded etcd, which bootstraps over HTTP.
-func (c *Cluster) save(ctx context.Context, override bool) error {
+func Save(ctx context.Context, config *config.Control, etcdConfig endpoint.ETCDConfig, override bool) error {
```

Comment:
Hmm, instead of being a Cluster method, this is now just a plain function that takes some fields that would normally be pulled from a pre-configured Cluster struct? Is there some reason this needs to be called from places that can't set up a Cluster first?

Reply:
Yes This is now explicitly called by secrets-encrypt in the server which does not have access to the cluster... this was discussed with Brian.

Reply:
got it, thanks!